src/compiler/printer/index.ts

```diff
@@ -0,0 +1,384 @@
+import { createPrinter, toString, group, line, softline, hardline, concat, indent, join, ifBreak } from './formatter';
+import { NodeKind, NodeFlags } from '../ast/node';
+
+export interface PrinterOptions {
+  tabWidth: number;
+  printWidth: number;
+  bracketSpacing: boolean;
+}
+
+export const enum PrettyPrint {}
+
+export function printSourceFile(node: any, options: PrinterOptions): any {
+  const printer = createPrinter(node.source);
+  return toString(printStatements(node, node), printer, options);
+}
+
+function printStatements(node: any, parentNode: any): any {
+  const kind = node.kind;
+
+  if (kind & NodeKind.IsIdentifier) {
+    return node.rawText;
+  }
+  if (kind & NodeKind.IsFunctionDeclaration) {
+    return printFunctionDeclarationOrExpression(node);
+  }
+
+  if (node.kind & NodeKind.IsTypescript) {
+    return printTypescript(node, parentNode);
+  }
+
+  switch (kind) {
+    case NodeKind.Script:
+      return printStatements(node.scriptBody, node);
+    case NodeKind.Module:
+      return printStatements(node.moduleBody, node);
+    case NodeKind.ScriptBody:
+    case NodeKind.ModuleBody:
+      return printList(node.statements, parentNode, printStatements);
+    case NodeKind.EmptyStatement:
+      return '';
+    case NodeKind.FormalParameterList:
+      return printFormalParameterList(node);
+    case NodeKind.FormalParameter:
+      return printFormalParameter(node);
+    case NodeKind.SingleNameBinding:
+      return printSingleNameBinding(node);
```

Comment:
@aladdin-add You need to **delete** this file. Not ready yet for open sourcing and belong to other private repo ;)